### ISSUE TITLE
Fails to compile in Elixir 1.15.2

### ISSUE THREAD

**Issue by plicjo** [knrz/towel#9]
```
==> postgrex
Compiling 68 files (.ex)
Generated postgrex app
error: undefined variable "package"
  /Users/joshuaplicque/projects/local_fare_jax/deps/towel/mix.exs:9: Towel.Mixfile.project/0

error: undefined variable "deps"
  /Users/joshuaplicque/projects/local_fare_jax/deps/towel/mix.exs:12: Towel.Mixfile.project/0

warning: function deps/0 is unused
  /Users/joshuaplicque/projects/local_fare_jax/deps/towel/mix.exs:25: Towel.Mixfile (module)
```

Compiling postgrex seems to fail in the Mix file.

It seems this issue might be resolved here: https://github.com/CyrusOfEden/towel/commit/6414289d7e093891a9665350213e71c54cd8f044#diff-dfa6f4ed74c90e5d4fda283d547d366586e690387289bcfd473e3fa5f9ace308R9-R12

I believe version 0.2.2 just needs to be published to Hex?

**Comment by plicjo** [comment]
Also, I used your `geocoder` project to much success on a project. Thank you so much for that simple, yet powerful library.

**Comment by plicjo** [comment]
Hex publishing guide: https://hex.pm/docs/publish

**Comment by evilmarty** [comment]
I can confirm that 45e6f733dc4a79ab3d5c8aee49e40d210d8ff1a2 works and, as mentioned above, just needs 0.2.2 to be published.

@ CyrusOfEden are you able to do that please?

In the meantime just add the following to your deps in `mix.exs`

```elixir
{:towel, github: "CyrusOfEden/towel", ref: "45e6f733dc4a79ab3d5c8aee49e40d210d8ff1a2", override: true},
```

**Comment by CyrusNuevoDia** [comment]
Hey @plicjo I get weird errors like this:

```
Compiling 5 files (.ex)
** (MatchError) no match of right hand side value: {:error, {{:EXIT, {:function_clause, [{:disk_log_sup, :start_link, [], [file: 'disk_log_sup.erl', line: 26]}, {:supervisor, :do_start_child_i, 3, [file: 'supervisor.erl', line: 420]}, {:supervisor, :do_start_child, 2, [file: 'supervisor.erl', line: 406]}, {:supervisor, :handle_start_child, 2, [file: 'supervisor.erl', line: 712]}, {:supervisor, :handle_call, 3, [file: 'supervisor.erl', line: 461]}, {:gen_server, :try_handle_call, 4, [file: 'gen_server.erl', line: 1113]}, {:gen_server, :handle_msg, 6, [file: 'gen_server.erl', line: 1142]}, {:proc_lib, :wake_up, 3, [file: 'proc_lib.erl', line: 251]}]}}, {:child, :undefined, :disk_log_sup, {:disk_log_sup, :start_link, []}, :permanent, false, 1000, :supervisor, [:disk_log_sup]}}}
    (kernel 9.0.2) disk_log_server.erl:142: :disk_log_server.ensure_started/0
    (kernel 9.0.2) disk_log_server.erl:52: :disk_log_server.open/1
    (stdlib 5.0.2) ets.erl:825: :ets.tab2file/3
    (mix 1.14.4) lib/mix/compilers/application_tracer.ex:250: Mix.Compilers.ApplicationTracer.write_manifest/1
    (mix 1.14.4) lib/mix/compilers/application_tracer.ex:41: Mix.Compilers.ApplicationTracer.prepare/2
    (mix 1.14.4) lib/mix/compilers/elixir.ex:150: Mix.Compilers.Elixir.compile/7
    (mix 1.14.4) lib/mix/state.ex:15: Mix.State.lock/2
    (mix 1.14.4) lib/mix/task.ex:421: anonymous fn/3 in Mix.Task.run_task/4
 ```
    
 When trying to compile the project. Investigating more...

**Comment by samhamilton** [comment]
@CyrusOfEden I just tested a compile locally with Elixir 1.15.4 and are not able to reproduce any errors

```
➜  towel git:(master) ✗ mix compile --warnings-as-errors
Compiling 5 files (.ex)
Generated towel app
```

**Comment by CyrusNuevoDia** [comment]
Awesome @samhamilton, I suspect it has something to do with my M1 Macbook. For now people can poiunt to the github.

**Comment by samhamilton** [comment]
@CyrusOfEden FYI also running a Macbook Air M1, no strange setup just installed via ASDF

```
elixir 1.15.4-otp-26
erlang 26.0.2
```

```
elixir -v
Erlang/OTP 26 [erts-14.0.2] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit]

Elixir 1.15.4 (compiled with Erlang/OTP 26)
```

**Comment by CyrusNuevoDia** [comment]
@samhamilton wish it was as easy as that, I've tried using asdf to do the installs and that's the error I was getting. Tried OTP 24, 25, and 26.

**Comment by evilmarty** [comment]
@CyrusOfEden why not setup a GitHub Action to build and publish the package?

**Comment by CyrusNuevoDia** [comment]
@evilmarty genius! setting this up now https://github.com/CyrusOfEden/towel/actions/runs/5633172705/job/15261793472

**Comment by CyrusNuevoDia** [comment]
Should be fixed with 0.2.2, published with the insight of @evilmarty, https://github.com/CyrusOfEden/towel/actions/runs/5633210719/job/15261859943
